packages/editor/src/internal-utils/parse-blocks.ts

```diff
@@ -296,22 +320,43 @@ function parseInlineObject({
     return undefined
   }
 
-  if (
-    inlineObject._type === context.schema.span.name ||
-    inlineObject._type === 'span' ||
-    // Respect the schema definition and don't parse inline objects that are not defined
-    !context.schema.inlineObjects.some(({name}) => name === inlineObject._type)
-  ) {
+  const schemaType = context.schema.inlineObjects.find(
+    ({name}) => name === inlineObject._type,
+  )
+
+  if (!schemaType) {
     return undefined
   }
 
+  // Validates all props on the inline object and only takes those that match
+  // the name of a field
+  const values = schemaType.fields.reduce<Record<string, unknown>>(
```

Comment:
Seems like there is some possiblity of DRY'ing between span and block? This and hidden stuff is duplicated 100% isnt it?

Reply:
Good point. I've now created a `parseObject(...)` function where most object parsing is handled.